### PR TITLE
Create auto-installation for textmode with sle-module-desktop-applications and sle-module-development-tools use libyui-rest-api

### PR DESCRIPTION
**Pull request by tinawang123**
Create auto-installation for textmode with sle-module-desktop-applications and sle-module-development-tools use libyui-rest-api

- Related ticket: https://progress.opensuse.org/issues/122353
- Verification run: https://openqa.suse.de/tests/10807396
  https://openqa.suse.de/tests/10724850
  https://openqa.suse.de/tests/10797330

### CODE REVIEW

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+    </products>
+  </software>
+  <ssh_import t="map">
+    <copy_config t="boolean">false</copy_config>
+    <import t="boolean">false</import>
+  </ssh_import>
+  <suse_register t="map">
+    <addons t="list">
+      <addon t="map">
+        <arch>{{ARCH}}</arch>
```
Comment (lemon-suse) [inline data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml:331]: For this profile is only for aarch64, we do not need the variable of ARCH here.

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+        <name>sle-module-desktop-applications</name>
+        <reg_code/>
+        <release_type>nil</release_type>
+        <version>{{VERSION}}</version>
+      </addon>
+      <addon t="map">
+        <arch>{{ARCH}}</arch>
```
Comment (lemon-suse) [inline data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml:338]: Same here and all later variable of ARCH.

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+      <package>autoyast2</package>
+    </packages>
+    <patterns t="list">
+      <pattern>apparmor</pattern>
+      <pattern>base</pattern>
+      <pattern>basesystem</pattern>
+      <pattern>basic_desktop</pattern>
+      <pattern>enhanced_base</pattern>
+      <pattern>minimal_base</pattern>
+      <pattern>x11</pattern>
```
Comment (lemon-suse) [inline data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml:316]: For textmode, no x11 needed, include the x11_yast.

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_s390x.xml [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+        <service>sshd</service>
+        <service>systemd-remount-fs</service>
+      </enable>
+    </services>
+  </services-manager>
+  <software t="map">
+    <install_recommended t="boolean">true</install_recommended>
+    <instsource/>
+    <packages t="list">
+      <package>xorg-x11-fonts</package>
```
Comment (lemon-suse) [inline data/autoyast_sle15/create_hdd/create_hdd_textmode_s390x.xml:336]: no x11 related package needed for textmode.

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml (hunk 1 above)
Comment (tinawang123) [reply data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml:331]: Thanks for your suggestion,   But I would like to keep it. It's better to use variable than constants.

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml (hunk 2 above)
Comment (tinawang123) [reply data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml:338]: Thanks for your suggestion,   But I would like to keep it. It's better to use variable than constants.

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml (hunk 3 above)
Comment (tinawang123) [reply data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml:316]: This x11 and x11_yast is needed by default pattern for textmode.

File: data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml (hunk 1 above)
Comment (lemon-suse) [reply data/autoyast_sle15/create_hdd/create_hdd_textmode_aarch64.xml:331]: Anyway, I don't think variable is needed for a consistent ARCH, while not a big thing here.

**Comment by hjluo** [comment]
LGTM

**Comment by rakoenig** [comment]
LGTM too.

**Pull request merged**